contrib/demo/clusters/kind/createKindClusters.sh

```diff
@@ -1,9 +1,34 @@
-#!/bin/bash -e
+#!/usr/bin/env bash
+
+set -euxo pipefail
 
 DEMO_ROOT="$(dirname "${BASH_SOURCE}")/../.."
+TMPDIR=$(mktemp -d)
+
+clusters="$@"
+if [[ $# -eq 0 ]]; then
+  clusters="us-east1 us-west1"
+fi
+
+for name in ${clusters}; do
+  cat > ${TMPDIR}/${name}.config << EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+name: ${name}
+networking:
+  apiServerAddress: "127.0.0.1"
+EOF
 
-kind create cluster --config ${DEMO_ROOT}/clusters/kind/us-east1.config --kubeconfig ${DEMO_ROOT}/clusters/kind/us-east1.kubeconfig
-sed -e 's/^/    /' ${DEMO_ROOT}/clusters/kind/us-east1.kubeconfig | cat ${DEMO_ROOT}/clusters/us-east1.yaml - > ${DEMO_ROOT}/clusters/kind/us-east1.yaml
+  kind delete cluster --name=${name} || true
+  kind create cluster --config ${TMPDIR}/${name}.config --kubeconfig ${TMPDIR}/${name}.kubeconfig
 
-kind create cluster --config ${DEMO_ROOT}/clusters/kind/us-west1.config --kubeconfig ${DEMO_ROOT}/clusters/kind/us-west1.kubeconfig
-sed -e 's/^/    /' ${DEMO_ROOT}/clusters/kind/us-west1.kubeconfig | cat ${DEMO_ROOT}/clusters/us-west1.yaml - > ${DEMO_ROOT}/clusters/kind/us-west1.yaml
+  cat > ${DEMO_ROOT}/clusters/${name}.yaml << EOF
+apiVersion: cluster.example.dev/v1alpha1
+kind: Cluster
+metadata:
+  name: ${name}
+spec:
+  kubeconfig: |
+EOF
+  sed -e 's/^/    /' ${TMPDIR}/${name}.kubeconfig >> ${DEMO_ROOT}/clusters/${name}.yaml
+done
```

Comment:
I say this, because this kubeconfig .yaml "Cluster" kind is something I can't consumer later down the road (I could strip it out again dynamically though).

Reply:
I completely agree